### ISSUE TITLE
Comments.

### ISSUE THREAD

**Issue by technetos** [jakogut/tinyvm#34]
Please comment this code.  It looks like a really excellent project for learning about virtual machines but people wont use it that way if its a pain to figure out whats going on in the code.

**Comment by vi** [comment]
Also README and/or repository description should state what type of a virtual machine is it. Is like hardware-level one like Qemu/virtualbox or language-level like JVM?

**Comment by SuperFola** [comment]
Why not adding some benchmarks also, to know about the performances of the VM on a few algorithms such as the fibbonacci sequence or the factorial ?

**Comment by stefanos82** [comment]
I agree with @vi. This project, even though very promising, it's not clear what its actual nature is.

I am interested in VM for programming languages to be honest with you, not for VM OS hosting emulators.

**Comment by technetos** [comment]
Id be totally willing to contribute to the docs if you set some up :smiley:

**Comment by SuperFola** [comment]
@stefanos82 by reading the code i think this is for a programming language

**Comment by stefanos82** [comment]
Nice! Thank you @SuperFola for letting me know.

**Comment by jakogut** [comment]
@technetos Some years ago, I started this project because I wanted to know more about how virtual machines work, so I wrote one. The syntax and function of the VM is described in the `SYNTAX` document.

@SuperFola There are example programs in `programs/tinyvm`, including Project Euler solutions, and a few comparable programs in C and x86 assembly.

I've been working on an off on a C library replacement that can host this project as well, which is linked as a submodule from [mlibc](https://github.com/jakogut/mlibc).

As far as further documentation goes, PRs are welcome.
